pq/pq.go

```diff
@@ -71,53 +82,44 @@ func MakeRoot() [SzRoot]byte {
 // start transactions. An error is returned if the delegate is nil, the queue
 // header is invalid, some settings are invalid, or if some IO error occurred.
 func New(delegate Delegate, settings Settings) (*Queue, error) {
+	const op = "pq/new"
+
 	if delegate == nil {
-		return nil, errNODelegate
+		return nil, txerr.Op(op).Of(InvalidParam).Msg("delegate must not be nil")
 	}
 
-	accessor, err := makeAccess(delegate)
-	if err != nil {
-		return nil, err
+	accessor, k := makeAccess(delegate)
+	if k != NoError {
+		return nil, txerr.Op(op).Of(k).Err()
 	}
 
-	q := &Queue{accessor: accessor}
-
 	pageSize := delegate.PageSize()
-	pagePool := newPagePool(pageSize)
+
+	q := &Queue{
+		accessor: accessor,
+		settings: settings,
+		pagePool: newPagePool(pageSize),
+	}
+
+	// use pointer address as ID for correlating error messages
+	q.id = queueID(uintptr(unsafe.Pointer(q)))
+	accessor.quID = q.id
 
 	rootBuf, err := q.accessor.ReadRoot()
 	if err != nil {
-		return nil, err
+		return nil, txerr.Op(op).Of(InitFailed).CausedBy(err).
+			Msg("failed to read queue header")
 	}
 
 	root := castQueueRootPage(rootBuf[:])
 	if root.version.Get() != queueVersion {
-		return nil, errInvalidVersion
+		return nil, txerr.Op(op).Of(InitFailed).
+			CausedBy(txerr.Of(QueueVersion).
+				Msgf("queue version %v", root.version.Get()))
 	}
 
 	tracef("open queue: %p (pageSize: %v)\n", q, pageSize)
 	traceQueueHeader(root)
-
-	tail := q.accessor.ParsePosition(&root.tail)
```

Comment:
If I get it right this was now move to the Writer.

Reply:
yeah. queue accessors are lazily instantiated by now.